examples/webgl2_buffergeometry_attributes_integer.html

```diff
@@ -0,0 +1,171 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>three.js WebGL 2 - buffergeometry - integer attributes</title>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="container"></div>
+		<div id="info"><a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> WebGL 2 - buffergeometry - integer attributes</div>
+
+		<script id="vertexShader" type="x-shader/x-vertex">#version 300 es
+			in int textureIndex;
+
+			flat out int vIndex; // "flat" indicates that the value will not be interpolated (required for integer attributes)
+			out vec2 vUv;
+
+			void main()	{
+
+				vIndex = textureIndex;
+				vUv = uv;
+
+				gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
+
+			}
+		</script>
+
+		<script id="fragmentShader" type="x-shader/x-fragment">#version 300 es
+			flat in int vIndex;
+			in vec2 vUv;
+
+			uniform sampler2D uTextures[ 3 ];
+
+			out vec4 outColor;
+
+			void main()	{
+
+				if ( vIndex == 0 ) outColor = texture( uTextures[ 0 ], vUv );
```

Comment:
Unfortunately it is not possible to do this:
```js
outColor = texture( uTextures[ vIndex ], vUv );
```
This throws: `sampler arrays indexed with non-constant expressions are forbidden`. So you can only use the integers for exact `==` comparisons.